test/priceUtils.js

```diff
@@ -26,5 +33,105 @@ contract("PriceOracle", function(accounts) {
       const price = 1000
       assert(await isPriceReasonable(exchange, targetTokenId, stableTokenId, price, acceptedPriceDeviationInPercentage))
     })
+    it("checks that bracket traders does not sell unprofitable for tokens with the same decimals", async () => {
+      const ERC20 = artifacts.require("DetailedMintableToken")
+      const token1 = await ERC20.new("WETH", 18)
+      const token2 = await ERC20.new("DAI", 18)
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token1.address, { from: accounts[0] })
+      await prepareTokenRegistration(accounts[0], exchange)
+      await exchange.addToken(token2.address, { from: accounts[0] })
+
+      const orders = [
+        {
+          // normal order selling for more than 120 DAI per WETH
+          user: "0xf888243aacb5626b520d0028371bad672a477fd8",
+          sellTokenBalance: new BN(0),
+          buyToken: 1,
+          sellToken: 2,
+          priceNumerator: new BN("1").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("115").mul(new BN(10).pow(new BN(18))),
+        },
+        {
+          // normal order selling for more than 120 DAI per WETH
+          user: "0xf888243aacb5626b520d0028371bad672a477fd8",
+          sellTokenBalance: new BN("125").mul(new BN(10).pow(new BN(18))),
+          buyToken: 2,
+          sellToken: 1,
+          priceNumerator: new BN("132").mul(new BN(10).pow(new BN(18))),
+          priceDenominator: new BN("1").mul(new BN(10).pow(new BN(18))),
+        },
+      ]
+
+      assert.equal(await checkNoProfitableOffer(orders[0], exchange, 1.0 / 120), true)
+      assert.equal(await checkNoProfitableOffer(orders[1], exchange, 120), true)
+    })
+    it("checks that bracket traders does not sell unprofitable for tokens with the different decimals", async () => {
```

Comment:
Some typos, my proposal:
```
detects unprofitable order for tokens with different decimals
```

Reply:
Typos are still there, this is because the same sentence was also used in the test immediately below, which was however changed. So maybe the two tests should have a different name?